### PR TITLE
Add range control

### LINKED ISSUE #134: Add support for range input
It would be great if you supported range inputs.

https://getbootstrap.com/docs/4.1/components/forms/#range

### PR DESCRIPTION
**Pull request by Munksgaard**
Very rough and initial commit of support for the bootstrap range control. Not
tested or verified in any way, other than a simple example I built separately
from this package.

Supposed to fix #134

### CODE REVIEW

File: src/Bootstrap/Form/Range.elm [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        Validation validation ->
+            { options | validation = Just validation }
+
+        Attrs attrs_ ->
+            { options | attributes = options.attributes ++ attrs_ }
+
+
+toAttributes : Options msg -> List (Html.Attribute msg)
+toAttributes options =
+    [ Attributes.class "form-control"
```
Comment (rundis) [inline src/Bootstrap/Form/Range.elm]: form-control-range

File: src/Bootstrap/Form/Range.elm [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    , validation : Maybe FormInternal.Validation
+    , attributes : List (Html.Attribute msg)
+    }
+
+
+{-| Create a range input.
+
+    Range.range
+        [ Range.id "myRange"
+        , Range.min -10
```
Comment (rundis) [inline src/Bootstrap/Form/Range.elm]: string no?

File: src/Bootstrap/Form/Range.elm [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        , Range.max 10
```
Comment (rundis) [inline src/Bootstrap/Form/Range.elm]: string?

File: src/Bootstrap/Form/Range.elm [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+{-| Shorthand for assigning an onInput handler for a range.
+-}
+onInput : Maybe (String -> msg) -> Option msg
+onInput toMsg =
+    OnInput toMsg
+
+
+{-| Option to set the min value for the range.
+-}
+min : String -> Option msg
```
Comment (rundis) [inline src/Bootstrap/Form/Range.elm]: Could maybe be number ?

File: src/Bootstrap/Form/Range.elm [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+min val =
+    Min val
+
+
+{-| Option to set the max value for the range.
+-}
+max : String -> Option msg
```
Comment (rundis) [inline src/Bootstrap/Form/Range.elm]: Could maybe be number ?

File: src/Bootstrap/Form/Range.elm [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+max val =
+    Max val
+
+
+{-| Option to set the step value for the range.
+-}
+step : String -> Option msg
```
Comment (rundis) [inline src/Bootstrap/Form/Range.elm]: Could be number ? If so, we should add an `anyStep` function as well (ref https://getbootstrap.com/docs/4.3/components/forms/#range-inputs)

Comment (Munksgaard) [reply src/Bootstrap/Form/Range.elm]: I'm not sure I understand what the anyStep function should do? 

With regards to `step`, `min` and `max` taking `number` as input, I don't think that's possible as there's no `number -> String` function as far as I'm aware. We could use floats, though. Would you prefer that?

Comment (rundis) [reply src/Bootstrap/Form/Range.elm]: from mdn "A string value of any means that no stepping is implied, and any value is allowed (barring other constraints, such as min and max)."

number -> string: Nah I think it's probably best to just stick with strings tbh.

Comment (Munksgaard) [reply src/Bootstrap/Form/Range.elm]: > from mdn "A string value of any means that no stepping is implied, and any value is allowed (barring other constraints, such as min and max)."

Good point! I assumed that was the default behavior, but I see that we'll need an anyStep.

Comment (Munksgaard) [reply src/Bootstrap/Form/Range.elm]: Although, perhaps now that we're making `step` take a `String` as input, perhaps it's not needed anyway?

File: src/Bootstrap/Form/Range.elm (hunk 1 above)
Comment (Munksgaard) [reply src/Bootstrap/Form/Range.elm]: Done

File: src/Bootstrap/Form/Range.elm (hunk 6 above)
Comment (rundis) [reply src/Bootstrap/Form/Range.elm]: Probably not worth it, as we accept strings we pretty much give up on type safety anyway.

Comment (Munksgaard) [reply src/Bootstrap/Form/Range.elm]: In that case I think I've resolved all the issues you pointed out.

**Comment by rundis** [comment]
TX for your contribution and patience. I'll add an example in https://github.com/rundis/elm-bootstrap.info forms page and take it from there!

**Comment by Munksgaard** [comment]
Thank you for your work :+1:

**Pull request merged**